src/compiler/builder.ts

```diff
@@ -889,16 +883,27 @@ namespace ts {
             }
         }
 
-        return {
+        let changeFileSet: ProgramBuildInfoFileId[] | undefined;
+        if (state.changedFilesSet.size) {
+            for (const path of arrayFrom(state.changedFilesSet.keys()).sort(compareStringsCaseSensitive)) {
+                (changeFileSet ||= []).push(toFileId(path));
+            }
+        }
+
+        const result: ProgramMultiFileEmitBuildInfo = {
             fileNames,
             fileInfos,
-            options: convertToProgramBuildInfoCompilerOptions(state.compilerOptions, relativeToBuildInfoEnsuringAbsolutePath),
+            options: convertToProgramBuildInfoCompilerOptions(state.compilerOptions, "affectsMultiFileEmitBuildInfo"),

```

Comment:
What's the string literal used for?

Reply:
Its the key of CommandLineOption that determines which option to be serialized in tsbuildinfo